### DISCUSSION TITLE
Is there a native way to detect android os?

### DISCUSSION THREAD

**Discussion by nextfool** [dotnet/runtime#60539] [Q&A]
so in the latest .net 6 rc2, we have a "OSPlatform" like this:
![image](https://user-images.githubusercontent.com/2167369/137687607-5677c524-bcac-4058-a16b-1e3cb0d1613f.png)

and in the OperratingSystem.cs,  IsOSPlatform like this:
![image](https://user-images.githubusercontent.com/2167369/137687784-11a7171c-6a76-4b24-9077-abb119e2c85d.png)

As you can see, there is NO option for android... Can any one please help me understand here, is this by desgin? How do you check if os is android?

Thanks.

**Comment by huoyaoyuan** [comment]
`OSPlatform` is somehow "deprecated". You can use `System.OperatingSystem.IsAndroid()`.

**Reply by John0King** [reply]
it seems a bug ?  because the `IsAndroid` have a TFM  android check , but the `IsOSPlatform`  missing this TFM check .(but android is using mono, so~)
https://github.com/dotnet/runtime/blob/2cf6b0d07309a3c19938f4c886fe13e8d5cc28ee/src/libraries/System.Private.CoreLib/src/System/OperatingSystem.cs#L89-L115

https://github.com/dotnet/runtime/blob/2cf6b0d07309a3c19938f4c886fe13e8d5cc28ee/src/libraries/System.Private.CoreLib/src/System/OperatingSystem.cs#L164-L173

`OsPlatform` is being used by `System.Runtime.InteropServices.RuntimeInfomation.IsOsPlatform` 

https://github.com/dotnet/runtime/blob/2cf6b0d07309a3c19938f4c886fe13e8d5cc28ee/src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/RuntimeInformation.cs#L49-L52

**Reply by huoyaoyuan** [reply]
It looks like, since it does check for ios and mac catalyst.

**Reply by huoyaoyuan** [reply]
OK, read the whole type and the answer is NOT a bug. Android falls under `Unix`, and is compared with `s_osPlatformName` which is readed by `Interop.Sys.GetUnixName()`. Not sure if it correctly returns android.
